File: mobile/lib/models/table.dart
class CafeTable {
  final String id;
  final String name;
  final int capacity;
  final String status; // 'available' | 'occupied' | 'reserved' (manual)
  final String liveStatus; // server-computed: 'available' | 'occupied' | 'reserved'
  final int openOrderCount;
  // Numeric identifier used in the customer QR ordering URL (/menu?table=N).
  // Null until an admin/manager assigns one (see QR Management screen).
  final int? number;

  CafeTable({
    required this.id,
    required this.name,
    required this.capacity,
    required this.status,
    required this.liveStatus,
    required this.openOrderCount,
    this.number,
  });

  bool get isOccupied => liveStatus == 'occupied';
  bool get isAvailable => liveStatus == 'available';
  bool get isReserved => liveStatus == 'reserved';
  bool get hasQrNumber => number != null;

  factory CafeTable.fromJson(Map<String, dynamic> json) {
    return CafeTable(
      id: json['_id'] as String,
      name: json['name'] as String? ?? '',
      capacity: (json['capacity'] as num?)?.toInt() ?? 4,
      status: json['status'] as String? ?? 'available',
      liveStatus: json['liveStatus'] as String? ?? json['status'] as String? ?? 'available',
      openOrderCount: (json['openOrderCount'] as num?)?.toInt() ?? 0,
      number: (json['number'] as num?)?.toInt(),
    );
  }
}

File: backend/src/models/Table.js
const mongoose = require('mongoose');

// Represents a physical dine-in table. `status` is mostly system-managed
// (occupied/available flip automatically as open orders come and go — see
// orderController) but can also be set manually to 'reserved' by staff.
const tableSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true }, // e.g. "Table 1"
    // Numeric identifier used in customer-facing QR ordering URLs, e.g.
    // /menu?table=5. Optional/sparse so existing tables keep working before
    // a number is assigned; enforced unique among tables that have one set.
    number: { type: Number, min: 1, unique: true, sparse: true },
    capacity: { type: Number, default: 4, min: 1 },
    status: { type: String, enum: ['available', 'occupied', 'reserved'], default: 'available' },
    active: { type: Boolean, default: true }, // soft delete/deactivate — never hard-delete a table with order history
  },
  { timestamps: true }
);

tableSchema.index({ active: 1, status: 1 });

module.exports = mongoose.model('Table', tableSchema);
